code/game/objects/items/mail.dm

```diff
@@ -0,0 +1,267 @@
+	// Destination tagging
+	if(istype(W, /obj/item/dest_tagger))
+		var/obj/item/dest_tagger/O = W
+
+		if(sortTag != O.currTag)
+			var/tag = uppertext(GLOB.TAGGERLOCATIONS[O.currTag])
+			to_chat(user, "<span class='notice'>*[tag]*</span>")
+			sortTag = O.currTag
+			playsound(loc, 'sound/machines/twobeep_high.ogg', 100, TRUE)
+
+/obj/item/mail/attack_self(mob/user)
+	if(recipient && user != recipient)
+		to_chat(user, "<span class='notice'>You can't open somebody else's mail! That's <em>illegal</em>!</span>")
+		return
+
+	to_chat(user, "<span class='notice'>You start to unwrap the package...</span>")
+	if(!do_after(user, 15, target = user))
+		return
+	user.temporarilyRemoveItemFromInventory(src, TRUE)
+	unwrap_contents()
+	for(var/mail_contents in contents)
+		var/atom/movable/AM = mail_contents
+		user.put_in_hands(AM)
+	playsound(src.loc, 'sound/items/poster_ripped.ogg', 50, TRUE)
+	qdel(src)
+
+/obj/item/mail/proc/unwrap_contents()
+	for(var/obj/object in GetAllContents())
+		SEND_SIGNAL(object, COMSIG_STRUCTURE_UNWRAPPED)
+
+/// Accepts a mob to initialize goodies for a piece of mail.
+/obj/item/mail/proc/initialize_for_recipient(mob/new_recipient)
+	recipient = new_recipient
+	name = "[initial(name)] for [recipient.real_name] ([recipient.job])"
+	var/list/goodies = generic_goodies
+
+	var/datum/job/this_job = SSjob.name_occupations[recipient.job]
+	if(this_job)
+		if(this_job.paycheck_department && department_colors[this_job.paycheck_department])
+			color = department_colors[this_job.paycheck_department]
+		var/list/job_goodies = this_job.get_mail_goodies()
+		if(LAZYLEN(job_goodies))
+			// certain roles and jobs (prisoner) do not receive generic gifts.
+			if(this_job.exclusive_mail_goodies)
+				goodies = job_goodies
+			else
+				goodies += job_goodies
+
+	for(var/iterator = 0, iterator < goodie_count, iterator++)
+		var/target_good = pickweight(goodies)
+		if(ispath(target_good, /datum/reagent))
+			var/obj/item/reagent_containers/target_container = new /obj/item/reagent_containers/glass/bottle(src)
+			target_container.reagents.add_reagent(target_good, target_container.volume)
+			target_container.name = "[target_container.reagents.reagent_list[1].name] bottle"
+			new_recipient.log_message("[key_name(new_recipient)] received reagent container [target_container.name] in the mail ([target_good])", LOG_GAME)
+		else
+			var/atom/movable/target_atom = new target_good(src)
+			new_recipient.log_message("[key_name(new_recipient)] received [target_atom.name] in the mail ([target_good])", LOG_GAME)
+
+	return TRUE
+
+/// Alternate setup, just complete garbage inside and anyone can open
+/obj/item/mail/proc/junk_mail()
+
+	var/obj/junk = /obj/item/paper/fluff/junkmail_generic
+	var/special_name = FALSE
+
+	if(prob(25))
+		special_name = TRUE
+		junk = pick(list(/obj/item/paper/pamphlet/gateway, /obj/item/paper/pamphlet/violent_video_games, /obj/item/paper/fluff/junkmail_redpill, /obj/effect/decal/cleanable/ash))
```

Comment:
as I mentioned earlier on discord, some anthrax mail (or at least mail coating in something mild, like itching powder) could be pretty funny to add to this list. remember that contact poison-laced paper is still a thing that we have in our code for some reason.

Reply:
For personal posterity I am not going to be coding anthrax.